### PR TITLE
chore: minor refactoring for protocols

### PR DESCRIPTION
**Pull request by danisharora099**
`BaseProtocol` housed `getPeer` as well as `getPeers` with very close naming.

- `getPeers` is used to find peers for a protocol, & filter according to discovery mechanism prioritization.
   - is being used by Filter, LightPush, Store
- `getPeer` simply found lowest latency peer for the protocol (while this was not required by the way it was being used after we introduced `getPeers`)
   - is being used by Metadata and Peer Exchange

This PR:
- removes the `getPeer` function and directly uses `peerStore.get(peerId)` to find a peer
   - this is done for peer-exchange and metadata -- this was being used to send a request to that peer
   - `peerId` is made mandatory on the params for peer-exchange query
- `getPeers` is used in all other protocols and the lowest latency sorting is added 

## Notes:
- test removed for `selectPeersForProtocol` & refactored test added
- [x] failing px test to be resolved by: https://github.com/waku-org/js-waku/pull/1773
- [x] failing multiple pubsub topic test: https://github.com/waku-org/js-waku/pull/1774

### CODE REVIEW

File: packages/tests/tests/utils.spec.ts [hunk 1]
```diff
@@ -121,148 +116,3 @@ describe("Util: toAsyncIterator: Filter", () => {
     expect(result.done).to.eq(true);
   });
 });
-
-const TestCodec = "test/1";
-
-describe("selectPeerForProtocol", () => {
```
Comment (adklempner) [inline packages/tests/tests/utils.spec.ts:127]: do all these tests need to be added elsewhere?

File: packages/core/src/lib/base_protocol.ts [hunk 2]
```diff
@@ -93,8 +83,19 @@ export class BaseProtocol implements IBaseProtocol {
...
-    // Filter the peers based on the specified criteria
-    return filterPeers(allPeersForProtocol, numPeers, maxBootstrapPeers);
+    // Filter the peers based on discovery & number of peers requested
+    const filteredPeers = await filterPeersByDiscovery(
+      allPeersForProtocol,
+      numPeers,
+      maxBootstrapPeers
+    );
+
+    const sortedFilteredPeers = await sortPeersByLatency(
```
Comment (fryorcraken) [inline packages/core/src/lib/base_protocol.ts:93]: Documentation of the function needs to be updated.

**Review by fryorcraken — COMMENTED** [review]
I am confused by the tests deletion

File: packages/core/src/lib/base_protocol.ts (hunk 2 above)
Comment (danisharora099) [reply packages/core/src/lib/base_protocol.ts:93]: Can you elaborate?

Doc for `sortPeersByLatency` was updated.
For the parent function `getPeers`, it has a generic definition that wasn't falsy. Updated it for more verbosity: 02aeab804782da1306627f73b14b180a5da80ed3

File: packages/tests/tests/utils.spec.ts (hunk 1 above)
Comment (danisharora099) [reply packages/tests/tests/utils.spec.ts:127]: right -- refactored tests and added here: b0fd0e36e8da8ad1c31e098106418505711af7d0

**Review by weboko — APPROVED** [review]
lgtm ✅

**Review by adklempner — APPROVED** [review]

File: packages/core/src/lib/base_protocol.ts (hunk 2 above)
Comment (danisharora099) [reply packages/core/src/lib/base_protocol.ts:102]: @fryorcraken merging this now
please let me know if you would prefer any follow up -- thanks!

**Pull request merged**